File: app/schemas/courses.py
from __future__ import annotations

from datetime import date

from sqlmodel import Field

from .base import AppSchema, PaginatedResponse


class CourseListItem(AppSchema):
    curso_id: str
    curso_nombre: str = ""
    curso_fecha: date | None = None


class CourseDetail(CourseListItem):
    pass


class CourseListResponse(PaginatedResponse):
    items: list[CourseListItem] = Field(default_factory=list)


__all__ = [
    "CourseDetail",
    "CourseListItem",
    "CourseListResponse",
]

File: frontend/src/types/courses.ts
export interface CourseListItem {
  curso_id: string
  curso_nombre: string
  curso_fecha: string | null
}

export type CourseDetail = CourseListItem

export interface CourseAttendeeItem {
  id: string
  nombre: string
  empresa: string
  confirmado: boolean
  observaciones: string
}

export interface CourseListResponse {
  total: number
  limit: number
  offset: number
  items: CourseListItem[]
}

export interface CourseAttendeeListResponse {
  total: number
  limit: number
  offset: number
  items: CourseAttendeeItem[]
}
